diff --git a/build/cmake/aom_optimization.cmake b/build/cmake/aom_optimization.cmake
index 8bffbe6..7e26f3a 100644
--- a/build/cmake/aom_optimization.cmake
+++ b/build/cmake/aom_optimization.cmake
@@ -214,6 +214,12 @@ function(test_nasm)
   execute_process(COMMAND ${CMAKE_ASM_NASM_COMPILER} -hf
                   OUTPUT_VARIABLE nasm_helptext)
 
+  if(NOT "${nasm_helptext}" MATCHES "-Ox")
+    execute_process(COMMAND ${CMAKE_ASM_NASM_COMPILER} -h -O
+                    OUTPUT_VARIABLE nasm_opt_helptext)
+    string(APPEND nasm_helptext "${nasm_opt_helptext}")
+  endif()
+
   if(NOT "${nasm_helptext}" MATCHES "-Ox")
     message(
       FATAL_ERROR "Unsupported nasm: multipass optimization not supported.")
